src/modules/pv_headers/pvh_func.c

```diff
@@ -383,51 +383,51 @@ int pvh_header_param_exists(struct sip_msg *msg, str *hname, str *hvalue)
 
 int pvh_remove_header_param(struct sip_msg *msg, int idx, str *hname, str *elements, str *toRemove)
 {
+	int notTarget, writtenChars;
 	int offset = 0;
 	int ret = -1;
 	char *next_token;
 	char *token;
 	char *result = (char*)pkg_malloc(elements->len - toRemove->len);
-	char *t = (char*)pkg_malloc(elements->len);
-	int maxSize = elements->len;
+	char *t = (char*)pkg_malloc(elements->len+1);
 
 	if (result == NULL || t == NULL)
 	{
 		PKG_MEM_ERROR;
 		goto clean;
 	}
 
-	snprintf(result, elements->len - toRemove->len, "%*s", elements->len - toRemove->len, "");
-	snprintf(t, elements->len+1, "%s", elements->s);
+	memset(result, 32, elements->len - toRemove->len);
```

Comment:
What is the purpose of filling `result` buffer with space (char code `32`)?

Reply:
Yes, here this is strange. No need of this line entirely.